### PR TITLE
fix(flags): block disabling flags with active dependencies

### PR DESCRIPTION
**Pull request by gustavohstrassburger**
## Problem

Previously when disabling a feature flag that other flags depend on, users would see a warning toast but could still proceed with the action.

Related to: https://posthoghelp.zendesk.com/agent/tickets/48384 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/52180)

image

## Changes

Added backend validation and frontend UI changes to prevent disabling feature flags that have active dependencies, making the system more opinionated about leaving flags in an invalid state.

## How did you test this code?

- Added new unit tests

## Publish to changelog?

no

### CODE REVIEW

File: posthog/api/test/test_feature_flag_dependency_disabling.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+        # Create dependent flag but make it inactive
+        self.create_flag("dependent_flag", dependencies=[base_flag.id], active=False)
+
+        # Should be able to disable base flag
+        response = self.client.patch(
+            f"/api/projects/{self.team.id}/feature_flags/{base_flag.id}/",
+            {"active": False},
+            format="json",
+        )
```
Comment (dustinbyrne) [inline posthog/api/test/test_feature_flag_dependency_disabling.py:99]: UX edge case here - presumably `dependent_flag` could be re-enabled without warning that the dependency chain cannot be evaluated

**Review by dustinbyrne — APPROVED** [review]
👍 👍 

One potential gap I thought of below, but maybe something more worth keeping in mind rather than urgent / needing a fix now. Up to you!

File: posthog/api/test/test_feature_flag_dependency_disabling.py (hunk 1 above)
Comment (gustavohstrassburger) [reply posthog/api/test/test_feature_flag_dependency_disabling.py:99]: you're right, and since this is easy enough it's worth doing it right now.
